mantidproject/mantid
# Rosace detector IDs for IN4 now set correctly

A bug from changes made in #17065 meant that the Rosace detectors get numbered incorrectly. This should fix that, add a test to sanity check the detector IDs and 

**To test:**

Load an IN4 NeXus file such as `/users/bush/ExternalData/Testing/Data/UnitTest/ILL/IN4/084446.nxs`. Check the detector IDs are 0 for the monitor, then run from 1 -> 396 for the other detectors.

Fixes #17756 .

_Does not need to be in the release notes._

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

Removed some code duplication that led to this bug. Also added a sanity check on the detector IDs, to check none have an invalid value.
